### PR TITLE
Update VMs, CMake to 3.20.1, CUDA to 11.3, and pwsh to 7.1.3

### PR DESCRIPTION
**Pull request by BillyONeal**
I tried updating to Ubuntu 20.04 as well but that's apparently not offered in a reasonable way on Azure right now.

### CODE REVIEW

**Comment by NancyLi1013** [comment]
The failures caused by `cudnn `depends on PR #17346.

**Comment by BillyONeal** [comment]
@NancyLi1013 Thanks for pointing those out!

**Comment by BillyONeal** [comment]
Looks like the many libvpx failures are https://github.com/microsoft/vcpkg/pull/17287/ ?

File: ports/libvpx/portfile.cmake [hunk 1]
```diff
@@ -99,14 +99,23 @@ if(VCPKG_TARGET_IS_WINDOWS AND NOT VCPKG_TARGET_IS_MINGW)
...
+    set(LIBVPX_OUTPUT_PREFIX "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/${LIBVPX_ARCH_DIR}")
     if(NOT DEFINED VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL "release")
-        file(INSTALL "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/${LIBVPX_ARCH_DIR}/Release/vpx.lib" DESTINATION "${CURRENT_PACKAGES_DIR}/lib")
-        file(INSTALL "${CURRENT_BUILDTREES_DIR}/${TARGET_TRIPLET}/${LIBVPX_ARCH_DIR}/Release/vpx/vpx.pdb" DESTINATION "${CURRENT_PACKAGES_DIR}/lib")
+        file(INSTALL "${LIBVPX_OUTPUT_PREFIX}/Release/vpx.lib" DESTINATION "${CURRENT_PACKAGES_DIR}/lib")
+        if (EXISTS "${LIBVPX_OUTPUT_PREFIX}/Release/vpx.pdb")
+            file(INSTALL "${LIBVPX_OUTPUT_PREFIX}/Release/vpx.pdb" DESTINATION "${CURRENT_PACKAGES_DIR}/lib")
+        else()
+            file(INSTALL "${LIBVPX_OUTPUT_PREFIX}/Release/vpx/vpx.pdb" DESTINATION "${CURRENT_PACKAGES_DIR}/lib")
+        endif()
```
Comment (JackBoosY) [inline ports/libvpx/portfile.cmake:109]: Sorry for that, but I think we still need to know what caused this.

**Review by JackBoosY — APPROVED** [review]

File: ports/chromaprint/vcpkg.json [hunk 2]
```diff
@@ -1,6 +1,7 @@
 {
   "name": "chromaprint",
   "version-string": "1.5.0",
```
Comment (NancyLi1013) [inline ports/chromaprint/vcpkg.json:3]:
```suggestion
  "version": "1.5.0",
```
We might use [version](https://github.com/microsoft/vcpkg/blob/master/docs/users/versioning.reference.md#version) here.

File: versions/c-/chromaprint.json [hunk 3]
```diff
@@ -1,5 +1,10 @@
 {
   "versions": [
+    {
+      "git-tree": "2532e2f07791dbd4e07913473cd77a2368c35332",
+      "version-string": "1.5.0",
```
Comment (NancyLi1013) [inline versions/c-/chromaprint.json:5]:
```suggestion
      "version": "1.5.0",
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by cenit** [comment]
you removed cudnn auto-download for recognized cuda versions. Wasn't it accepted that it was ok in other PRs?

This is removing a feature otherwise which might be the base on which someone has built something... (me)

**Comment by BillyONeal** [comment]
> you removed cudnn auto-download for recognized cuda versions. Wasn't it accepted that it was ok in other PRs?

It was accepted in other PRs because from original contribution it looked like downloading any other closed source library, which would normally be OK. Unfortunately, it has been brought to our attention that this behavior is both (1) contrary to cudnn's license terms, and (2) contrary to anaconda.org's terms of service. As a result I feel forced to remove it. I did ask some nvidia folks to give us better options: https://twitter.com/MalwareMinigun/status/1384335333249097732

> This is removing a feature otherwise which might be the base on which someone has built something... (me)

I'm sorry :(

**Comment by jacobkahn** [comment]
I've removed this functionality from the `nccl` port as well in https://github.com/microsoft/vcpkg/pull/17431. It would be great to get a CUDA 11.3-compatible NCCL version on the CI machines as well.

**Comment by PhoebeHui** [comment]
@BillyONeal, we noticed that many PRs failed with following failures on osx pipeline machine, could you please help take a look? 

```
Unable to locate executable file: 'pwsh'. Please verify either the file path exists or the file can be found within a directory specified by the PATH environment variable. Also check the file mode to verify the file is executable.
```

**Comment by BillyONeal** [comment]
> @BillyONeal, we noticed that many PRs failed with following failures on osx pipeline machine, could you please help take a look?
> 

Unfortunately I know very little about how the osx stuff works; Nicole was working on that

File: ports/dartsim/portfile.cmake [hunk 4]
```diff
@@ -9,6 +9,7 @@ vcpkg_from_github(
     HEAD_REF release-6.9
     PATCHES 1478.patch
             1497.patch
+	    disable_unit_tests_examples_and_tutorials.patch
 )
 
 vcpkg_configure_cmake(
```
Comment (NancyLi1013) [inline ports/dartsim/portfile.cmake:15]: The format should be like this:

```
 OPTIONS 
    -DDART_VERBOSE=ON
    -DDART_MSVC_DEFAULT_OPTIONS=ON
    -DDART_SKIP_DOXYGEN=ON
    -DDART_SKIP_FLANN=ON
    -DDART_SKIP_IPOPT=ON
    -DDART_SKIP_NLOPT=ON
    -DDART_SKIP_OPENGL=ON
    -DDART_SKIP_pagmo=ON
```

Comment (NancyLi1013) [reply ports/dartsim/portfile.cmake:15]: `file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/dartsim RENAME copyright)
`

should be updated as

`file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)`

File: ports/libudis86/portfile.cmake [hunk 5]
```diff
@@ -11,8 +11,8 @@ file(COPY ${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt DESTINATION ${SOURCE_PATH})
 vcpkg_find_acquire_program(PYTHON2)
 
 vcpkg_execute_required_process(
-    COMMAND ${PYTHON2} ${SOURCE_PATH}/scripts/ud_itab.py  ${SOURCE_PATH}/docs/x86/optable.xml ${SOURCE_PATH}/libudis86/
-    WORKING_DIRECTORY SOURCE_PATH
+    COMMAND "${PYTHON2}" "${SOURCE_PATH}/scripts/ud_itab.py" "${SOURCE_PATH}/docs/x86/optable.xml" "${SOURCE_PATH}/libudis86/"
+    WORKING_DIRECTORY "${SOURCE_PATH}"
     LOGNAME python-${TARGET_TRIPLET}-generate-sources
 )
```
Comment (NancyLi1013) [inline ports/libudis86/portfile.cmake:18]: The format of options for debug should be like this:
```
OPTIONS_DEBUG
    -DDISABLE_INSTALL_HEADERS=ON
    -DDISABLE_INSTALL_TOOLS=ON
```

`file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/libudis86 RENAME copyright)`

 should be updated as

`file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)`

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
On x86-windows:

`libhv `failed with this error:

```
CMake Error at CMakeLists.txt:60 (configure_file):
  Permission denied
```
Maybe this is related with parallel issue?

`kf5windowsystem` failed with this error:

```
FAILED: src/KF5WindowSystem_autogen/timestamp 
cmd.exe /C "cd /D D:\buildtrees\kf5windowsystem\x86-windows-dbg\src && D:\downloads\tools\cmake-3.20.1-windows\cmake-3.20.1-windows-i386\bin\cmake.exe -E cmake_autogen D:/buildtrees/kf5windowsystem/x86-windows-dbg/src/CMakeFiles/KF5WindowSystem_autogen.dir/AutogenInfo.json Debug && D:\downloads\tools\cmake-3.20.1-windows\cmake-3.20.1-windows-i386\bin\cmake.exe -E touch D:/buildtrees/kf5windowsystem/x86-windows-dbg/src/KF5WindowSystem_autogen/timestamp && D:\downloads\tools\cmake-3.20.1-windows\cmake-3.20.1-windows-i386\bin\cmake.exe -E cmake_transform_depfile Ninja gccdepfile D:/buildtrees/kf5windowsystem/src/v5.75.0-98cbac325e.clean D:/buildtrees/kf5windowsystem/src/v5.75.0-98cbac325e.clean/src D:/buildtrees/kf5windowsystem/x86-windows-dbg D:/buildtrees/kf5windowsystem/x86-windows-dbg/src D:/buildtrees/kf5windowsystem/x86-windows-dbg/src/KF5WindowSystem_autogen/deps D:/buildtrees/kf5windowsystem/x86-windows-dbg/CMakeFiles/d/ac0678512fc3ae64732cd27db69d826af10d6262969c164f975967658f984c1c.d"
ninja: build stopped: subcommand failed.
```

**Comment by NancyLi1013** [comment]
`dimcli `failed on x64-windows-static and x64-widnows-static-md with this error:

```
-- Install configuration: "Debug"
         -- Installing: D:/packages/dimcli_x64-windows-static-md/debug/lib/dimcli.lib
         CMake Error at cmake_install.cmake:45 (file):
           file INSTALL cannot find
           "D:/buildtrees/dimcli/x64-windows-static-md-dbg/dimcli.dir/Debug/dimcli.pdb":
           File exists.
```
`wangle` failed on x64-windows and x64-windows-static-md with this error:
```
D:\installed\x64-windows-static-md\include\fizz/protocol/FizzBase.h(139): fatal error C1001: Internal compiler error.
(compiler file 'msc1.cpp', line 1588)
```

`paraview ` failed on x64-windows-static with this error:

```
FAILED: Qt/ApplicationComponents/pqApplicationComponents_autogen/timestamp 
cmd.exe /C "cd /D D:\buildtrees\paraview\x64-windows-static-dbg\Qt\ApplicationComponents && D:\downloads\tools\cmake-3.20.1-windows\cmake-3.20.1-windows-i386\bin\cmake.exe -E cmake_autogen D:/buildtrees/paraview/x64-windows-static-dbg/Qt/ApplicationComponents/CMakeFiles/pqApplicationComponents_autogen.dir/AutogenInfo.json Debug && D:\downloads\tools\cmake-3.20.1-windows\cmake-3.20.1-windows-i386\bin\cmake.exe -E touch D:/buildtrees/paraview/x64-windows-static-dbg/Qt/ApplicationComponents/pqApplicationComponents_autogen/timestamp && D:\downloads\tools\cmake-3.20.1-windows\cmake-3.20.1-windows-i386\bin\cmake.exe -E cmake_transform_depfile Ninja gccdepfile D:/buildtrees/paraview/src/4fb6e6800f-6b3165e6c7.clean D:/buildtrees/paraview/src/4fb6e6800f-6b3165e6c7.clean/Qt/ApplicationComponents D:/buildtrees/paraview/x64-windows-static-dbg D:/buildtrees/paraview/x64-windows-static-dbg/Qt/ApplicationComponents D:/buildtrees/paraview/x64-windows-static-dbg/Qt/ApplicationComponents/pqApplicationComponents_autogen/deps D:/buildtrees/paraview/x64-windows-static-dbg/CMakeFiles/d/9f52e9f0cf771bc230acb38a0b83aad4a0e8fbcb2a50c4bba7e0ead5db381cdc.d"
```

**Comment by NancyLi1013** [comment]
`libidn2 `failed on osx with this error:

```
autoreconf: export WARNINGS=
autoreconf: Entering directory '.'
autoreconf: running: autopoint --force
autoreconf: running: aclocal --force -I m4 -I unistring/m4
autoreconf: configure.ac: tracing
autoreconf: running: glibtoolize --copy --force
autoreconf: configure.ac: not using Intltool
autoreconf: running: gtkdocize --copy
Can't exec "gtkdocize": No such file or directory at /usr/local/Cellar/autoconf/2.71/share/autoconf/Autom4te/FileUtils.pm line 293.
autoreconf: error: gtkdocize failed with exit status: 2
```
This should be fixed in PR #17419, please see https://github.com/microsoft/vcpkg/pull/17419#issuecomment-824819810 and https://github.com/microsoft/vcpkg/pull/17419#issuecomment-825719305.

Seems you forgot to add `forest:x64-osx=fail` in ci.baseline.txt.

**Comment by BillyONeal** [comment]
>For forest, seems you forgot to add forest:x64-osx=fail in ci.baseline.txt.

I didn't see it before because this PR didn't trigger a world-rebuild that attempted to build it on osx until I did the cmake version update.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
All checks have passed now.

File: scripts/azure-pipelines/windows/utility-prefix.ps1 [hunk 6]
```diff
@@ -0,0 +1,125 @@
+# Copyright (c) Microsoft Corporation.
```
Comment (strega-nil) [inline scripts/azure-pipelines/windows/utility-prefix.ps1:1]: This should probably be a `utilities.psm1`, like in macOS, but not blocking

**Review by strega-nil — APPROVED** [review]

File: scripts/azure-pipelines/windows/utility-prefix.ps1 (hunk 6 above)
Comment (BillyONeal) [reply scripts/azure-pipelines/windows/utility-prefix.ps1:1]: We can't shove .psm1s through `Invoke-AzVmRunCommand` though; we're taking this script and find and replacing `# REPLACE WITH UTILITY-PREFIX.ps1` with it.

**Comment by JackBoosY** [comment]
Great! Thanks Billy!

**Pull request merged**